field-testing/src/lib.rs

```diff
@@ -91,6 +91,27 @@ pub fn test_two_adic_coset_zerofier<F: TwoAdicField>() {
     }
 }
 
+pub fn test_two_adic_generator_consistency<F: TwoAdicField>() {
+    let log_n = F::TWO_ADICITY;
+    let g = F::two_adic_generator(log_n);
+    for i in 0..log_n {
+        assert_eq!(g.exp_power_of_2(i), F::two_adic_generator(log_n - i));
+    }
+}
+
+pub fn test_ef_two_adic_generator_consistency<
+    F: TwoAdicField,
+    EF: TwoAdicField + ExtensionField<F>,
+>() {
+    let log_n = F::TWO_ADICITY;
+    for bits in 1..log_n {
```

Comment:
Similarly here, can this be expanded to `0..=log_n`?

Reply:
I think we only need to check the case of `F::TWO_ADICITY` if both F and EF have passed the `test_two_adic_generator_consistency`. If we add the loop, the two tests will somewhat overlap. 

I have changed to a single assertion for the clear purpose of the test. WDYT?